### PR TITLE
directoryToSemverishObject function

### PR DESCRIPTION
**Pull request by patrickocoffeyo**
This PR introduces the following:
 - A `directoryToSemverishObject function that takes an absolute path to a directory, and walks the directory, and creates a semverish object from the directory's structure and files.

## Steps to test
- [ ] Checkout this branch
- [ ] Run `yarn test`.
- [ ] Note that the `directoryToSemverishObject.js` test passes.

## NOTE
@thebruce a couple of things:
 - [x] I pulled in your semverish test help directory and semverishObject from `semverish`. Should this be a PR against that repository instead?
 - [x] I need to finish writing tests for `directoryToSemverishObject.js`. Will work on that next, but I wanted to get this in front of your eyes early.

### CODE REVIEW

**Comment by thebruce** [comment]
@patrickocoffeyo thanks patrick, eventually yes - and if you'll be so kind i will work with you on semverist - its not quite ready to do this step async over there yet. so we'll let this be our test bed.

**Comment by patrickocoffeyo** [comment]
@thebruce sweet! I'll finish writing tests for this and then go from there!

**Comment by patrickocoffeyo** [comment]
@thebruce tests are fixed up, should be at 100% coverage.

File: src/directoryToSemverishObject.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          _.set(semverishObject, objPath, {});
+        }
+        else {
+          const p = path.parse(item.path);
+          // Remove the file name from the object path, and replace the path
+          // separator with dot notation.
+          objPath = `${objPath.replace(p.base, '').replace(pathSep, '.')}['${p.name}']`;
+
+          // Grab this file's content.
+          let objPathValue = fs.readFileSync(item.path, { encoding: 'utf8' });
```
Comment (thebruce) [inline src/directoryToSemverishObject.js]: Could we use the async version of fs? like i ended up using in the defaults loader? https://github.com/schemepunk/schema-got-swagger/blob/master/src/getDefaults.js#L130

Comment (patrickocoffeyo) [reply src/directoryToSemverishObject.js]: Yeah totally! I'll cook that up

**Comment by thebruce** [comment]
Very much needed. thank you @patrickocoffeyo

**Comment by thebruce** [comment]
Version [1.0.0](https://github.com/schemepunk/schema-got-swagger/releases/tag/1.0.0) has been released.

**Pull request merged**